### PR TITLE
[Impeller] Use direct tessellation geometry for simple clip shapes

### PR DESCRIPTION
**Pull request by flar**
This work leverages the recent work to directly tessellate the simple primitives for rendering and uses their geometry classes for equivalent clip primitives.

### CODE REVIEW

**Comment by flar** [comment]
There are 2 new goldens, but a couple of the old tests have "off by 1" pixels on the edges of the RRect clips that I've left for reviewers to see.

Perf data, using my primitive basher. Note that I didn't think this would show much improvement at all since I have to do:

```
  Save()
  ClipFoo(...)
  DrawRect(bounds, paint)
  Restore()
```

for each operation, but on my M1 MBPro:

5k clip primitives on MBP M1 Max

**Comment by flar** [comment]
I was going to provide numbers for my Pixel 6a, but the text on the Performance Overlay seems to have disappeared.

File: impeller/aiks/canvas.cc [hunk 1]
```diff
@@ -448,23 +448,34 @@ void Canvas::ClipRect(const Rect& rect, Entity::ClipOperation clip_op) {
...
+  auto geometry = Geometry::MakeOval(bounds);
+  auto& cull_rect = transform_stack_.back().cull_rect;
+  if (clip_op == Entity::ClipOperation::kIntersect &&                      //
+      cull_rect.has_value() &&                                             //
+      geometry->CoversArea(transform_stack_.back().transform, *cull_rect)  //
+  ) {
+    return;  // This clip will do nothing, so skip it.
+  }
+
+  ClipGeometry(geometry, clip_op);
```
Comment (jonahwilliams) [inline impeller/aiks/canvas.cc:461]:
```suggestion
  ClipGeometry(std::move(geometry), clip_op);
```

**Review by jonahwilliams — APPROVED** [review]
LGTM!

File: impeller/aiks/canvas.cc (hunk 1 above)
Comment (flar) [reply impeller/aiks/canvas.cc:461]: Skipping to avoid having to modify more method signatures to go along with this change...

**Pull request merged**